diff -uNr qemu-2.9.1/configure qemu-2.9.1.mod/configure
--- qemu-2.9.1/configure        2024-07-14 10:59:46.792259485 +0530
+++ qemu-2.9.1.mod/configure    2024-07-14 11:03:36.858612780 +0530
@@ -2402,6 +2402,7 @@
 fi

 has_libgcrypt_config() {
+  return 0
     if ! has "libgcrypt-config"
     then
        return 1
@@ -2426,10 +2427,7 @@
         # Debian has remove -lgpg-error from libgcrypt-config
         # as it "spreads unnecessary dependencies" which in
         # turn breaks static builds...
-        if test "$static" = "yes"
-        then
-            gcrypt_libs="$gcrypt_libs -lgpg-error"
-        fi
+        gcrypt_libs="$gcrypt_libs -lgpg-error"
         libs_softmmu="$gcrypt_libs $libs_softmmu"
         libs_tools="$gcrypt_libs $libs_tools"
         QEMU_CFLAGS="$QEMU_CFLAGS $gcrypt_cflags"
@@ -3717,6 +3715,7 @@
 if compile_prog "" "" ; then
   signalfd=yes
 fi
+signalfd=no

 # check if eventfd is supported
 eventfd=no
@@ -5255,6 +5254,7 @@
 if test "$darwin" = "yes" ; then
   echo "CONFIG_DARWIN=y" >> $config_host_mak
 fi
+ivshmem=no

 if test "$aix" = "yes" ; then
   echo "CONFIG_AIX=y" >> $config_host_mak
@@ -5378,9 +5378,9 @@
 if test "$posix_fallocate" = "yes" ; then
   echo "CONFIG_POSIX_FALLOCATE=y" >> $config_host_mak
 fi
-if test "$sync_file_range" = "yes" ; then
-  echo "CONFIG_SYNC_FILE_RANGE=y" >> $config_host_mak
-fi
+# if test "$sync_file_range" = "yes" ; then
+#   echo "CONFIG_SYNC_FILE_RANGE=y" >> $config_host_mak
+# fi
 if test "$fiemap" = "yes" ; then
   echo "CONFIG_FIEMAP=y" >> $config_host_mak
 fi
