diff -uNr qemu-2.9.1/linux-user/strace.c qemu-2.9.1.mod/linux-user/strace.c
--- qemu-2.9.1/linux-user/strace.c  2017-09-07 21:58:14.000000000 +0530
+++ qemu-2.9.1.mod/linux-user/strace.c  2024-07-14 12:27:55.694328722 +0530
@@ -1,7 +1,7 @@
 #include "qemu/osdep.h"
 #include <sys/ipc.h>
 #include <sys/msg.h>
-#include <sys/sem.h>
+#include <linux/sem.h>
 #include <sys/shm.h>
 #include <sys/select.h>
 #include <sys/mount.h>
