diff -uNr qemu-2.9.1/tcg/i386/tcg-target.inc.c qemu-2.9.1.mod/tcg/i386/tcg-target.inc.c
--- qemu-2.9.1/tcg/i386/tcg-target.inc.c  2017-09-07 21:58:14.000000000 +0530
+++ qemu-2.9.1.mod/tcg/i386/tcg-target.inc.c  2024-07-14 17:48:48.713710159 +0530
@@ -1534,7 +1534,11 @@
 # include <asm/prctl.h>
 # include <sys/prctl.h>

-int arch_prctl(int code, unsigned long addr);
+# include <linux/unistd.h>
+static int arch_prctl(int code, unsigned long addr)
+{
+    return syscall(__NR_arch_prctl, code, addr);
+}

 static int guest_base_flags;
 static inline void setup_guest_base_seg(void)